refactor: minio -> s3 (#23)
falcondev-oss/github-actions-cache-server: docs/content/2.storage-drivers/s3.md

---
title: S3
description: This storage driver stores the cache in a S3 compatible storage, incl. AWS S3 and MinIO.
---

Driver: `s3`

This storage driver stores the cache in a S3 compatible storage, incl. AWS S3 and MinIO.

## Configuration

### `docker-compose` example

```yaml [docker-compose.yml]
version: '3.9'

services:
  cache-server:
    image: ghcr.io/falcondev-oss/github-actions-cache-server:latest
    ports:
      - '3000:3000'
    environment:
      CACHE_SERVER_TOKEN: random_token
      BASE_URL: http://localhost:3000

      STORAGE_DRIVER: s3
      S3_BUCKET: gh-actions-cache
      S3_ACCESS_KEY: access_key
      S3_SECRET_KEY: secret_key
      S3_ENDPOINT: minio
      S3_PORT: '9000'
      S3_USE_SSL: 'false'
    volumes:
      - cache-data:/app/data

  minio:
    image: quay.io/minio/minio
    ports:
      - '9000:9000'
    environment:

>       S3_ROOT_USER: access_key
>       S3_ROOT_PASSWORD: secret_key

Reviewer: ```suggestion
      MINIO_ROOT_USER: access_key
      MINIO_ROOT_PASSWORD: secret_key
```

---

Driver: `s3`

This storage driver stores the cache in a S3 compatible storage, e.g. AWS S3 or MinIO.

## Configuration

### `docker-compose` MinIO example

```yaml [docker-compose.yml]
version: '3.9'

services:
  cache-server:
    image: ghcr.io/falcondev-oss/github-actions-cache-server:latest
    ports:
      - '3000:3000'
    environment:
      CACHE_SERVER_TOKEN: random_token
      BASE_URL: http://localhost:3000

      STORAGE_DRIVER: s3
      S3_BUCKET: gh-actions-cache
      S3_ACCESS_KEY: access_key
      S3_SECRET_KEY: secret_key

      S3_ENDPOINT: minio
      S3_PORT: '9000'
      S3_USE_SSL: 'false'
    volumes:
      - cache-data:/app/data

  minio:
    image: quay.io/minio/minio
    ports:
      - '9000:9000'
    environment:
      MINIO_ROOT_USER: access_key
      MINIO_ROOT_PASSWORD: secret_key

volumes:
  cache-data:
```

### `docker-compose` AWS S3 example

```yaml [docker-compose.yml]
version: '3.9'

services:
  cache-server:
    image: ghcr.io/falcondev-oss/github-actions-cache-server:latest
    ports:
      - '3000:3000'
    environment:
      CACHE_SERVER_TOKEN: random_token
      BASE_URL: http://localhost:3000

      STORAGE_DRIVER: s3

>       S3_BUCKET: gh-actions-cache

Reviewer: Add host, port and ssl defaults for AWS

      S3_ACCESS_KEY: access_key

>       S3_SECRET_KEY: secret_key

Reviewer: ```suggestion
      S3_SECRET_KEY: secret_key
      
      S3_ENDPOINT: s3.amazonaws.com
      S3_PORT: '443'
      S3_USE_SSL: 'true'
```

    volumes:
      - cache-data:/app/data

volumes:
  cache-data:
```

### Environment Variables

Don't forget to set the `STORAGE_DRIVER` environment variable to `s3` to use the S3 storage driver.

#### `S3_BUCKET`

Example: `gh-actions-cache`

The name of the S3 bucket used for storage.

#### `S3_ACCESS_KEY`

Example: `access_key`

The access key for S3 storage.

#### `S3_SECRET_KEY`

Example: `secret_key`

The secret key for S3 storage.

#### `S3_ENDPOINT`

Example: `s3.amazonaws.com`, `minio`

> The endpoint hostname for S3 storage. Omit or use `s3.amazonaws.com` for AWS S3.

Reviewer: ```suggestion
The endpoint hostname for S3 storage.
```

#### `S3_REGION`

Example: `us-west-1`

> The region for AWS S3. Unused for MinIO.

Reviewer: ```suggestion
The region for AWS S3. Not needed with MinIO.
```

#### `S3_PORT`

Example: `443`, `9000`

> The port S3 storage is running on. Omit or use `443` for AWS S3.

Reviewer: ```suggestion
The port S3 storage is running on.
```

#### `S3_USE_SSL`

> - Default: `false`

Reviewer: ```suggestion
Example: `false`
```

> Whether to use SSL for S3 storage connections. Omit or use `true` for AWS S3.

Reviewer: ```suggestion
Whether to use SSL for S3 storage connections.
```
